pandas/core/indexes/base.py

```diff
@@ -292,54 +294,59 @@ def __new__(
 
         name = maybe_extract_name(name, data, cls)
 
+        if dtype is not None:
+            dtype = pandas_dtype(dtype)
+        if "tz" in kwargs:
+            tz = kwargs.pop("tz")
+            validate_tz_from_dtype(dtype, tz)
+            dtype = tz_to_dtype(tz)
+
         if isinstance(data, ABCPandasArray):
             # ensure users don't accidentally put a PandasArray in an index.
             data = data.to_numpy()
 
+        data_dype = getattr(data, "dtype", None)
```

Comment:
can you rename to data_dtype, i find this spelling odd :-<

Reply:
hah, typo